File: zebra_print.dart
import 'dart:io';
import 'dart:core';
import 'dart:typed_data';

void main() async {

  print("Name:");
  String? name = stdin.readLineSync();

  print("User ID:");
  String? userID = stdin.readLineSync();

  print("Ticket Number:");
  String? ticketNumber = stdin.readLineSync();

  print("Call (Y/N:");
  String? callUser = stdin.readLineSync();

  String? today = "08/20/2021";

  final x = "^XA"
      "^CFE,30^FO12,30^FD$name^FS"
  "^FO310,120^BQN,2,6,Q,7^FD$userID^FS"
  "^FO12,80^FD$userID^FS"
  "^FO12,130^FD$ticketNumber^FS"
  "^FO12,180^FDCall: $callUser^FS"
  "^FO12,230^FD$today^FS"
  "^XZ";


  // connect to the socket server
  final socket = await Socket.connect('172.19.69.188', 9100);
  print('Connected to: ${socket.remoteAddress.address}:${socket.remotePort}');

  // listen for responses from the server
  socket.listen(

    // handle data from the server
        (Uint8List data) {
      final serverResponse = String.fromCharCodes(data);
      print('Server: $serverResponse');
    },

    // handle errors
    onError: (error) {
      print(error);
      socket.destroy();
    },

    // handle server ending connection
    onDone: () {
      print('Server left.');
      socket.destroy();
    },
  );

  await sendLabel(socket, x);

}

Future<void> sendLabel(Socket socket, String message) async {
  print('Client: $message');
  socket.write(message);
  await Future.delayed(Duration(seconds: 2));
}

File: zebra_print.py
import sys
import socket
from datetime import date

today = date.today()
today = today.strftime("%m/%d/%Y")

mysocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

# set printer ip and port variables 
# printer_ip = input('Printer IP Address: ')
# printer_port = input('Port: ')

try:
    printer = {'ip': printer_ip, 'port': printer_port}
except NameError:
    print('printer_ip and printer_port variables need to be defined before running script')
    sys.exit(1)

def print_label():
    # printer = input('Choose printer: ')
    name = input('Name: ')
    ticket_number = input('Ticket Number: ')
    user_id = input('UserID: ')
    call = input('Call user - (Y/N): ')

    # must be three spaces between the userid and ^FD for the QR Code to print each time
    x = "^XA" \
        "^FO300,100^BQ,2,6^FD   " + user_id + "^FS" \
         "^CFE,40^FO12,50^FD" + name + "^FS" \
        "^FO12,90^FD" + user_id + "^FS" \
        "^FO12,130^FD" + ticket_number + "^FS" \
        "^FO12,170^FDCall: " + call + "^FS" \
        "^FO12,210^FD" + today + "^FS" \
        "^XZ"

    x = x.encode()

    try:
        mysocket.connect((printer['ip'], printer['port']))  # connecting to host
        mysocket.send(x)  # using bytes
        mysocket.close()  # closing connection
    except:
        print("Error with the connection")


def main():
    print_label()


if __name__ == "__main__":
    main()

